guides/octane-auth-torii-with-github.md

```diff
@@ -0,0 +1,666 @@
+[Back to Main README](../README.md)
+
+# Full GitHub Authorization in Ember Octane with `torii-provider`
+
+This guide is an update and reworking of [the guide provided by
+`ember-simple-auth`](https://github.com/simplabs/ember-simple-auth/blob/master/guides/auth-torii-with-github.md),
+but it uses Octane syntax and eschews jQuery. It also uses a production-ready
+gatekeeper server instead of an `http-mock` server to provide the token
+exchange. 
+
+The [basic flow of authenticating with
+GitHub](https://developer.github.com/apps/building-oauth-apps/authorizing-oauth-apps/)
+deserves an outline that is personalized for Ember:
+
+1. When the user clicks the login button, the Ember application contacts
+   `https://github.com/login/oauth/authorize` with your app’s `client_id`.
+   GitHub presents a new window asking the user to authorize access to their
+   GitHub account from your app. The user grants this.
+2. GitHub redirects back to your app with an `authorizationCode`. This is
+   proof of authorization, but your app still doesn’t know _who_ the user is
+   (so it cannot connect to a `User` model, for example).
+3. Your app sends the `authorizationCode` to a gatekeeper server you maintain,
+   which in turn `POST`s the code to
+   `https://github.com/login/oauth/access_token`. In development, this can be
+   mocked. In production, it must be a separate server, about which more
+   below, as we will use a separate server throughout.
+4. GitHub responds to the `POST` with an `access_token`, which your gatekeeper
+   server forwards to your app.
+5. Your app uses the `access_token` in an `Authorization: Bearer
+   <access_token>` header request to, say, `https://api.github.com/user`,
+   which returns the GitHub user’s user info, at which point you can match the
+   username with your `User` model or whatever. 
+
+As with the original `ember-simple-auth` guide, the approach here mirrors any
+OAuth2 explicit grant flow used also by Facebook, Google, etc. 
+
+You can find a sequence diagram for the flow
+[here](http://github.com/simplabs/ember-simple-auth/guides/assets/esa-initial-flow.txt.pn://github.com/simplabs/ember-simple-auth/blob/master/guides/assets/esa-initial-flow.txt.png).
```

Comment:
It seems the link is not pointing to the image yet - do you intend to do the following?

```suggestion
[here](https://github.com/simplabs/ember-simple-auth/blob/master/guides/assets/esa-initial-flow.txt.png).
```

Reply:
yes thank you.